odl/discr/lp_discr.py

```diff
@@ -387,14 +385,9 @@ def __repr__(self):
             use_uniform = False
 
         if use_uniform:
-            if isinstance(self.dspace, Fn):
-                impl = 'numpy'
-                default_dtype = np.float64
-            elif isinstance(self.dspace, CudaFn):
-                impl = 'cuda'
-                default_dtype = np.float32
-            else:  # This should never happen
-                raise RuntimeError('unable to determine data space impl')
+            impl = self.dspace.impl
+            default_dtype = self.dspace.default_dtype(RealNumbers())
```

Comment:
How about default ordering? You wrote earlier that `'F'` should be default for CUDA. Maybe that information should be brought in by the plug-in?